### ISSUE TITLE
VFP related is broken for flow_ebos

### ISSUE THREAD

**Issue by GitPaean** [OPM/opm-simulators#1248]
Not sure since when yet.

**Comment by GitPaean** [comment]
The reason is the following code in `SimulatorFullyImplicitBlackoilEbos.hpp`.

```
436         well_model.init(phaseUsage,
437                         activePhases,
438                         /*vfpProperties=*/nullptr,
439                         gravity,
440                         legacyDepth_,
441                         legacyPoreVolume_,
442                         rateConverter_.get(),
443                         globalNumCells,
444                         grid());
```

**Comment by GitPaean** [comment]
The running was fixed by #1249, the outputting of THP is still missing for unknown reason.

**Comment by GitPaean** [comment]
The results of `flow_ebos` is not as good as `flow_legacy` in some cases we have.  Will check against the 2017.04 release to find some clues when finding time.

**Comment by GitPaean** [comment]
It is broken again. We need to fix it and create a running test to protect it.

**Comment by totto82** [comment]
Could you share some details about what model this is broken for?

**Comment by GitPaean** [comment]
I think the vfp related pointers are NULL again, so it should be easy to fix. It will be a little more complicated if it is the result not matching well after fixing the pointers. I will  send you a message.

**Comment by GitPaean** [comment]
One difficulty and cause that makes the flow vfp is not easy for some example is that, in the `flow`, there are some places we calculate the RS and RV based on the rates and pressure within wellbore. When the rates are too roughly initialized, the RS and RV will not be reasonable. In the `flow_legacy`, the rs and rv from reservoir blocks are used, then it is kind of avoids the RS and RV problem resulting from too roughly initialized well rates.

**Comment by GitPaean** [comment]
We understand this issue now better. It is closed by #1477.
